docs/articles/deployments.md

```diff
@@ -0,0 +1,237 @@
+
+**Known caveats:** Marathon has a cap on the number of concurrently running deployments. This is controlled by `--max_running_deployments` [command line argument](command-line-arguments.md). In some rare occasions, one can have the maximum number of running deployments achieved so that canceling one is not possible (since a new deployment has to be created). In this case, canceling using `?force=true` parameter aka. [forced deployment](deployments.md#forced-deployments) should be used.
+
+## Forced deployments
+Let's get back to our `product/service/rails-app` service deployment that is now "stuck" with slightly exaggerated resource requirements. We realize that 1k CPUs is too much and decide to update the `product/service/rails-app` service definition with reasonable CPU requirements. However, when submitting new service definition, Marathon will respond with `HTTP 209 (conflict)`:
+```
+{
+    "message": "App is locked by one or more deployments. Override with the option '?force=true'",
+    "deployments": [
+        {
+            "id": "3aa7316a-7888-427b-9af3-4179a7791c79"
+        }
+    ]
+}
+```
+This is due to already existing (stuck) deployment that affects our service definition. In general, Marathon will only allow one deployment at a time affecting any service definition (or service definition group). Let's try to submit our corrected definition again, however, this time using `?force=true` parameter (`POST /v2/apps?force=true`). This time, the result is `HTTP 200(ok)` and we receive a new deployment Id back. Under the hood, Marathon will cancel the previous deployment and start the new one. The new deployment will reconcile existing tasks (if any), check their service definition version, kill old ones if necessary and start new tasks.
+
+**Known caveats:** Due to its "destructive potential" usage of forced deployments is discouraged. A forced deployment will cancel any existing conflicting deployments (there might be more than one, see [group deployments](deployments.md#group-deployments)) potentially killing more tasks than the user intended.
+
+## Stopping a deployment
+In some rare cases a deployment might be stuck and canceling it is not an option (maybe because some tasks were already partially started and the user doesn't want to kill them). In this case, a deployment might be stopped. Stopping a deployment (as opposed to canceling a deployment) **will not return Marathon to the previous state**. Whichever tasks were already started will continue to run, the rest will not be launched. This operation will leave the service in an inconsistent state e.g. only 3/5 running tasks or 3 tasks running with the newest service definition version and 2 with the old one. **Stopping deployments should be a measure of last resort. Its usage is strongly discouraged**.
```

Comment:
```suggestion
In some rare cases a deployment might be stuck and canceling it is not an option (for example because some tasks were already partially started and the user doesn't want to kill them). In this case, the deployment might be stopped. Stopping a deployment (as opposed to canceling a deployment) **will not return Marathon to the previous state**. Whichever tasks were already started will continue to run, the rest will not be launched. This operation will leave the service in an inconsistent state e.g. only 3/5 running tasks or 3 tasks running with the newest service definition version and 2 with the old one. **Stopping deployments should be a measure of last resort. Its usage is strongly discouraged**.
```